code/game/objects/structures/windoor_assembly.dm

```diff
@@ -66,253 +67,214 @@ obj/structure/windoor_assembly/Destroy()
+		user.visible_message("[user] is cutting the wires from [src].", "You start to cut the wires from [src].")
+
+		if(do_after(user, src, 40))
+			if(gcDestroyed)
+				return
+			to_chat(user, "<span class='notice'>You cut \the [name] wires!</span>")
+			new /obj/item/stack/cable_coil(get_turf(user), 1)
+			wired = FALSE
+			update_name()
+
+
+	//Adding airlock electronics for access. Step 6 complete.
+	if(istype(W, /obj/item/weapon/circuitboard/airlock) && anchored)
+		var/obj/item/weapon/circuitboard/airlock/AE = W
+		if(AE.icon_state =="door_electronics_smoked")
+			to_chat(user, "<span class='notice'>\The [AE.name] is too damaged to work.</span>")
+			return
+		if(user.drop_item(AE, src)) // To prevent you using the airlock electronics on 2 windoors at once.
+			playsound(src, 'sound/items/Screwdriver.ogg', 100, 1)
+			user.visible_message("[user] installs [AE] into [src].", "You start to install [AE] into [src].")
+
+			if(do_after(user, src, 40))
+				if(gcDestroyed)
 					return
-				user.visible_message("[user] wires the windoor assembly.", "You start to wire the windoor assembly.")
-
-				if(do_after(user, src, 40))
-					if(!src)
-						return
-					CC.use(2)
-					to_chat(user, "<span class='notice'>You wire the windoor!</span>")
-					src.state = "02"
-					if(src.secure)
-						src.name = "Secure Wired Windoor Assembly"
-					else
-						src.name = "Wired Windoor Assembly"
+				to_chat(user, "<span class='notice'>You've installed [AE]!</span>")
+				electronics = AE
+				electronics.installed = TRUE
+				update_name()
+
+	//Screwdriver to remove airlock electronics. Step 6 undone.
+	if(isscrewdriver(W) && (anchored && electronics))
+		playsound(src, 'sound/items/Screwdriver.ogg', 100, 1)
+		user.visible_message("[user] removes the [electronics] from [src].", "You start to uninstall [electronics] from [src].")
+
+		if(do_after(user, src, 40))
+			if(gcDestroyed)
+				return
+			to_chat(user, "<span class='notice'>You've removed [electronics]!</span>")
+			var/obj/item/weapon/circuitboard/airlock/ae
+			ae = electronics
+			ae.installed = FALSE
+			electronics = null
+			ae.forceMove(loc)
+			update_name()
+
+
+	//Crowbar to complete the assembly, Step 7 complete.
+	if(iscrowbar(W) && anchored)
+		if(!wired)
+			to_chat(usr, "<span class='rose'>\The [name] is missing wires.</span>")
+			return
+		if(!electronics)
+			to_chat(usr, "<span class='rose'>\The [name] is missing electronics.</span>")
+			return
+		usr << browse(null, "window=windoor_access")
+		playsound(src, 'sound/items/Crowbar.ogg', 100, 1)
+		user.visible_message("[user] is prying [src] into the frame.", "You start prying [src] into the frame.")
+
+		if(do_after(user, src, 40))
+			if(gcDestroyed)
+				return
+			var/spawn_type = secure ? secure_type : windoor_type
+			var/obj/machinery/door/window/windoor = new spawn_type(loc)
+			to_chat(user, "<span class='notice'>You finish the [windoor.name]!</span>")
+			windoor.dir = dir
+			windoor.fingerprints += fingerprints
```

Comment:
Isn't there an addfingerprints or transferfingerprints function that can be used here?

Reply:
That's only for mobs, we're copying the fingerprints from one object to another

Reply:
https://github.com/vgstation-coders/vgstation13/blob/Bleeding-Edge/code/game/atoms.dm#L739

Reply:
Now THAT's the thing